File: Numerical integration/CompoundSimpson.jl
function comsimpson(f::Function,a,b,n)
    h=(b-a)/n
    nodes =Array{Float64}(undef,n+1)
    for i in 1:n+1
        nodes[i] =a+(i-1)*h
    end
    sum=f(a)+f(b)
    for i in 3:2:n-1
        sum=sum+2*f(nodes[i])
    end
    for i in 2:2:n
        sum=sum+4*f(nodes[i])
    end
    return (sum*h/3)
end
comsimpson(x->x*log(x),1,2,12)

File: Numerical integration/CompoundSimpson.py
import numpy as np
def comSimpson(f,a,b,n):
    h=(b-a)/n
    xi=np.array([a+(i+1)*h for i in range(n)])
    sum=f(a)+f(b)
    for i in range(1,n,2):
        sum=sum+4*f(xi[i])
    for i in range(2,n,2):
        sum=sum+2*f(xi[i])
    return (sum*h/3)
def f(x):
    if x==0: return 1
    else: return np.sin(x)/x
print(comSimpson(f,0,1,20))
